MoreLinq/Traverse.cs

```diff
@@ -0,0 +1,97 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Felipe Sateler. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Generates a sequence by traversing an element according to the <paramref name="nextSteps"/> generator function. 
```

Comment:
There seems to be general vocabulary/wording/model confusion between `TraverseBreadthFirst` and `TraverseDepthFirst` being generators or tree traversal algorithms. I think you need to settle on one and then have consistent wording and naming. I'd say that since the operators speak of traversing, the tree model is what gets evoked first (at least did for me and I'm imagining it'll be the same for others). But then `seed` and `nextSteps` evoke a generator model. Within the implementation, the loop `child` variable (albeit being a local variable) evokes the tree once again. The doc summaries talk in terms of both, which made it confusing to read and review.